pruning/export_test.go

```diff
@@ -0,0 +1,12 @@
+package pruning
```

Comment:
why is this a test file?

Reply:
Since tests e.g. `pruning_test` are a separate package from the main package e.g. `pruning`, we cannot test unexported functions. At the same time, we might not want to export some functions just for tests.

Go `..._test.go` files do not get compiled into normal binaries. 

So we can use this `export_test.go` trick to export some unexported functions for tests only.

Reply:
Yes, I see. But that's still not an idiomatic approach IMO. I'd expect to see tests here. So I'd recommend either:

1. Renaming this to `util_test.go` or something similar to indicate that it's for test utilities but no actual tests, OR
2. Rename the package to `pruning_internal` and the file can be named whatever at that point.

Reply:
> But that's still not an idiomatic approach IMO

Thanks for the suggestions. However, I've seen this idiom in multiple codebases. E.g. https://redhatinsights.github.io/insights-operator-cli/packages/export_test.html

To me, something that has `util` in its name would contain some functions that are used by tests while `export` would be used exactly for this pattern

I'm not feeling strongly about this. Just wanted to explain how I see it. So If you still would like me to change, please let me know

> Rename the package to pruning_internal and the file can be named whatever at that point.

If we do this, then this pattern wouldn't work. It only works when the following are satisfied:
- the tests are located in [the same package](https://github.com/golang/go/issues/11198#issuecomment-306195848) with the exports
- the export file (`export_test.go`) is the same package as the where the unexported items are located (`pruning` in our case). If the package name is different, we wouldn't have access to these unexported items.
- the export file is suffixed with `_test.go` so that it is not compiled for production use

Reply:
I'm sorry, I'm not following. This file files has zero tests and you're saying it's contents aren't used solely for testing utilities?

Reply:
They are used solely for testing. I was just trying to point out that `export_test.go` is a common idiom for exporting private constants, variables, and functions from the regular package to the test package.

On the other hand, something that is named `util_test.go`, is commonly used for utility functions /variables in tests that are not exported from the regular package.

Both idioms are used to be utilized in tests. They just do it differently. `export_test.go` enables accessing unexported constants, variables, and functions of the regular package in the test package. `util_test.go` and similar provide utilities that are unrelated to the regular package and are solely used in tests

Reply:
I see. This is fine then 👍